### ISSUE TITLE
`yield from` loses the delegate's return type when either function is annotated with a type alias

### ISSUE THREAD

**Issue by tanagumo** [astral-sh/ty#4230]
## Summary

`yield from` infers the delegate's return type correctly only when both the delegating and the delegated function spell out `Generator[...]` directly. If either annotation goes through a `type` alias of `Generator`, the result of the `yield from` expression becomes `Unknown`.

The alias does not have to be generic — a concrete `type C = Generator[int, Any, None]` loses it too.

No diagnostic is emitted, so the effect is a silent loss of checking rather than a false positive.

## Reproduction

```python
from collections.abc import Generator
from typing import Any

type Step[E, R] = Generator[E, Any, R]


def callee_raw() -> Generator[int, Any, str]:
    yield 1
    return "x"


def callee_alias() -> Step[int, str]:
    yield 1
    return "x"


def caller_raw_from_raw() -> Generator[int, Any, None]:
    a = yield from callee_raw()
    reveal_type(a)      # ty: str          (expected)


def caller_raw_from_alias() -> Generator[int, Any, None]:
    b = yield from callee_alias()
    reveal_type(b)      # ty: Unknown      (expected str)


def caller_alias_from_raw() -> Step[int, None]:
    c = yield from callee_raw()
    reveal_type(c)      # ty: Unknown      (expected str)


def caller_alias_from_alias() -> Step[int, None]:
    d = yield from callee_alias()
    reveal_type(d)      # ty: Unknown      (expected str)
```

## Expected

`str` in all four cases.

## Actual

| callee | caller | ty 0.0.66 | pyright 1.1 | mypy 2.3.0 | pyrefly 1.2.0 |
|---|---|---|---|---|---|
| direct | direct | `str` | `str` | `str` | `str` |
| alias | direct | **`Unknown`** | `str` | `str` | `str` |
| direct | alias | **`Unknown`** | `str` | `str` | `str` |
| alias | alias | **`Unknown`** | `str` | `str` | `str` |

## Versions

- ty 0.0.66
- Reproduced on Python 3.12, 3.13 and 3.14 (`--python-version`); the `type` statement in the reproduction requires 3.12+

**Comment by carljm** [comment]
Excellent timing! The fix for this was just merged in https://github.com/astral-sh/ruff/pull/27577 and released in ty 0.0.70.
